### DISCUSSION TITLE
Error "Couldn't determine Mocha version"

### DISCUSSION THREAD

**Discussion by sergefokin** [cypress-io/cypress#26381] [Questions and Help]
Hello.
I tried to setup reporting from cypress to testrail but after run command like 
$ npx cypress run --reporter junit --reporter-options "mochaFile=reports/TEST-[hash].xml"
I received an error "Couldn't determine Mocha version"
the same problem was in https://github.com/cypress-io/cypress/issues/4602 but it was marked as resolved.
I tried solutions from that topic but without success.

**Comment by portyu9** [comment]
Hi @sergefokin 

The command you are using is valid Cypress syntax, and Cypress's `junit` reporter is built in:

```sh
npx cypress run \
  --reporter junit \
  --reporter-options "mochaFile=reports/TEST-[hash].xml"
```

You do **not** need to install `mocha-junit-reporter` separately just to use Cypress's built-in JUnit output.

The `"Couldn't determine Mocha version"` message has historically come from reporter/version-detection code, particularly when third-party JUnit or multi-reporter packages are also installed or configured.

I would first simplify the setup:

1. Remove any separately configured `mocha-junit-reporter`, `mocha-multi-reporters`, or `cypress-multi-reporters` if you do not actually need them.
2. Reinstall dependencies cleanly.
3. Run only the built-in Cypress JUnit command above.
4. Check whether `reports/TEST-....xml` is produced.

If the tests finish and the XML file is generated, then `"Couldn't determine Mocha version"` is only reporter console output and is not what is preventing the TestRail import.

If the Cypress process actually exits because of it, the next useful information would be the full stack trace together with the Cypress version and reporter-related dependencies from `package.json`.

If you intentionally need a third-party multi-reporter setup, update those reporter packages and make sure they resolve Mocha from the same project in which Cypress is executing rather than adding another copy of Mocha manually.

Please mark this answer as accepted if it helped, thank you.
